molgenis-questionnaires/src/main/java/org/molgenis/questionnaires/QuestionnairePluginController.java

```diff
@@ -117,42 +126,47 @@ public String showThanks(@PathVariable("name") String name, Model model, HttpSer
 		return "view-thanks";
 	}
 
-	private Entity createQuestionnaireEntity(EntityType emd, QuestionnaireStatus status)
+	private synchronized Entity createQuestionnaireEntity(EntityType entityType, QuestionnaireStatus status,
+			String name)
 	{
-		Entity entity = entityManager.create(emd, POPULATE);
-		entity.set(OwnedEntityType.OWNER_USERNAME, SecurityUtils.getCurrentUsername());
-		entity.set(QuestionnaireMetaData.ATTR_STATUS, status.toString());
-		dataService.add(emd.getName(), entity);
-
+		Entity entity = findQuestionnaireEntity(name);
+		if (entity == null)
+		{
+			entity = entityManager.create(entityType, POPULATE);
+			entity.set(OWNER_USERNAME, getCurrentUsername());
+			entity.set(ATTR_STATUS, status.toString());
+			dataService.add(entityType.getName(), entity);
+		}
 		return entity;
 	}
 
-	private Questionnaire toQuestionnaireModel(Entity entity, EntityType emd)
+	private Questionnaire toQuestionnaireModel(Entity entity, EntityType entityType)
 	{
-		QuestionnaireStatus status = QuestionnaireStatus.valueOf(entity.getString(QuestionnaireMetaData.ATTR_STATUS));
-		return new Questionnaire(emd.getName(), emd.getLabel(languageService.getCurrentUserLanguageCode()), status,
-				emd.getDescription(languageService.getCurrentUserLanguageCode()), entity.getIdValue());
+		QuestionnaireStatus status = QuestionnaireStatus.valueOf(entity.getString(ATTR_STATUS));
+		return new Questionnaire(entityType.getName(),
+				entityType.getLabel(languageService.getCurrentUserLanguageCode()), status,
+				entityType.getDescription(languageService.getCurrentUserLanguageCode()), entity.getIdValue());
 	}
 
 	private Entity findQuestionnaireEntity(String name)
 	{
-		return dataService.findOne(name, EQ(OwnedEntityType.OWNER_USERNAME, SecurityUtils.getCurrentUsername()));
+		return dataService.findOne(name, EQ(OWNER_USERNAME, getCurrentUsername()));
 	}
 
-	public String getThankYouText(String questionnaireName)
+	private String getThankYouText(String questionnaireName)
 	{
 		return runAsSystem(() -> thankYouTextService.getThankYouText(questionnaireName));
 	}
 
-	public static class Questionnaire
+	private static class Questionnaire
```

Comment:
Took me one hour to find this easter (halloween?) egg.
Private classes cannot be read in the FreeMarker template.
Please make public again.